subprojects/docs/src/docs/userguide/inspectingDependencies.adoc

```diff
@@ -84,3 +84,20 @@ Every Gradle project provides the task `dependencyInsight` to render the so-call
     <output args="-q dependencyInsight --dependency commons-codec --configuration scm"/>
 </sample>
 ++++
+
+[[sec:dependency_declaration_reasons]]
+=== Justifying dependency declarations with custom reasons
+
+When you declare a <<declaring_dependencies,dependency>> dependency or a <<sec:dependency_constraints,dependency constraint>>, you can provide a custom reason for the declaration. This makes your build script and the dependency insight report easier to comprehend.
+
+++++
+<sample id="dependencyReason" dir="userguide/dependencyManagement/inspectingDependencies/dependencyReason" title="Giving a reason why a certain modules version is used in a dependency declaration">
```

Comment:
> a certain modules version

Does not read well, either the 's' from 'modules' needs to go or this has to be rephrased.

Reply:
removed the 's' and rephrased